src/main/java/com/hermesworld/ais/galapagos/kafka/impl/ConnectedKafkaCluster.java

```diff
@@ -137,14 +138,12 @@ public String getId() {
         ConfigResource res = new ConfigResource(Type.TOPIC, topicName);
         Config config = new Config(configs);
 
-        return toCompletableFuture(
-                this.adminClient
-                        .createTopics(
-                                Collections.singleton(new NewTopic(topicName, topicCreateParams.getNumberOfPartitions(),
-                                        (short) topicCreateParams.getReplicationFactor())))
-                        .all()).thenCompose(
-                                o -> toCompletableFuture(
-                                        adminClient.alterConfigs(Collections.singletonMap(res, config)).all()));
+        // TODO numPartitions should be configurable for Galapagos metadata topics
+        NewTopic newTopic = new NewTopic(topicName, 3, (short) topicCreateParams.getReplicationFactor());
```

Comment:
Nope, 3 ist wrong. Must be `topicCreateParams.getNumberOfPartitions()` (can be set via UI). Only for the Galapagos metadata topics (where this code is copied from), it is currently 3 as a constant.